src/compiledOutput.ts

```diff
@@ -27,10 +35,15 @@ export class DTSDocumentProvider implements vscode.TextDocumentContentProvider {
     activate(ctx: vscode.ExtensionContext) {
         ctx.subscriptions.push(vscode.workspace.registerTextDocumentContentProvider('devicetree', this));
         ctx.subscriptions.push(
-            vscode.commands.registerCommand(DTSDocumentProvider.COMMAND, (uri?: dts.DTSCtx | vscode.Uri, options?: vscode.TextDocumentShowOptions) => {
-                if (uri instanceof dts.DTSCtx) {
-                    uri = uri.files.pop()?.uri;
-                } else if (!uri && vscode.window.activeTextEditor?.document.languageId === 'dts') {
+            vscode.commands.registerCommand(DTSDocumentProvider.COMMAND, (arg?: dts.DTSCtx | vscode.Uri | BuildConfiguration, options?: vscode.TextDocumentShowOptions) => {
+                let uri: vscode.Uri;
+                if (arg instanceof dts.DTSCtx) {
+                    uri = arg.files.pop()?.uri;
+                } else if (arg && !(arg instanceof vscode.Uri)) {
+                    // uri is BuildConfiguration from nordicsemiconductor.nrf-connect
+                    const ctx = dts.parser.ctx(arg.conf.devicetree.id);
+                    uri = ctx.files.pop()?.uri;
+                } else if (!arg && vscode.window.activeTextEditor?.document.languageId === 'dts') {
                     uri = vscode.window.activeTextEditor?.document.uri;
                 }
```

Comment:
Missing an else here, setting `uri = arg;`. Alternatively, `arg instanceof vscode.Uri` could be the first check, and the instanceof check on line 44 can be removed